From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bharat Nihalani <bnihalani@nvidia.com>
Date: Tue, 13 Jan 2026 08:43:32 +0000
Subject: [PATCH 181/240] NVIDIA: SAUCE: soc/tegra: cbb: Fix build with ACPI
 disabled

CONFIG_ACPI is not used for embedded-linux and hence needs to be
disabled to optimize the kernel. Fix build errors in this driver
with CONFIG_ACPI disabled.

Bug 5786711

Change-Id: I1f80cde05360f6700c49f525f874d34643e93c63
Signed-off-by: Bharat Nihalani <bnihalani@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3527047
Reviewed-by: svcacv <svcacv@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
Reviewed-by: Jon Hunter <jonathanh@nvidia.com>
(cherry picked from commit e05745d7d36986d507a6bfb45f07394cb1d70f25)

diff --git a/drivers/soc/tegra/cbb/tegra234-cbb.c b/drivers/soc/tegra/cbb/tegra234-cbb.c
index 921550996723..8ac6344049d0 100644
--- a/drivers/soc/tegra/cbb/tegra234-cbb.c
+++ b/drivers/soc/tegra/cbb/tegra234-cbb.c
@@ -1,7 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0
+// SPDX-FileCopyrightText: Copyright (c) 2021-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 /*
- * Copyright (c) 2021-2025, NVIDIA CORPORATION. All rights reserved
- *
  * The driver handles Error's from Control Backbone(CBB) version 2.0.
  * generated due to illegal accesses. The driver prints debug information
  * about failed transaction on receiving interrupt from Error Notifier.
@@ -1659,6 +1658,7 @@ static const struct of_device_id tegra234_cbb_dt_ids[] = {
 };
 MODULE_DEVICE_TABLE(of, tegra234_cbb_dt_ids);
 
+#ifdef CONFIG_ACPI
 struct tegra234_cbb_acpi_uid {
 	const char *hid;
 	const char *uid;
@@ -1675,9 +1675,15 @@ static const struct tegra234_cbb_acpi_uid tegra234_cbb_acpi_uids[] = {
 };
 
 static const struct
-tegra234_cbb_fabric *tegra234_cbb_acpi_get_fabric(struct acpi_device *adev)
+tegra234_cbb_fabric *tegra234_cbb_acpi_get_fabric(struct device *dev)
 {
 	const struct tegra234_cbb_acpi_uid *entry;
+	struct acpi_device *adev = ACPI_COMPANION(dev);
+
+	if (!adev) {
+		dev_err(dev, "acpi_device is NULL\n");
+		return NULL;
+	}
 
 	for (entry = tegra234_cbb_acpi_uids; entry->hid; entry++) {
 		if (acpi_dev_hid_uid_match(adev, entry->hid, entry->uid))
@@ -1686,6 +1692,7 @@ tegra234_cbb_fabric *tegra234_cbb_acpi_get_fabric(struct acpi_device *adev)
 
 	return NULL;
 }
+#endif /* CONFIG_ACPI */
 
 static const struct acpi_device_id tegra241_cbb_acpi_ids[] = {
 	{ "NVDA1070" },
@@ -1743,19 +1750,16 @@ static int tegra234_cbb_probe(struct platform_device *pdev)
 	unsigned long flags = 0;
 	int err;
 
-	if (pdev->dev.of_node) {
+	if (pdev->dev.of_node)
 		fabric = of_device_get_match_data(&pdev->dev);
-	} else {
-		struct acpi_device *device = ACPI_COMPANION(&pdev->dev);
-		if (!device)
-			return -ENODEV;
-
-		fabric = tegra234_cbb_acpi_get_fabric(device);
-		if (!fabric) {
-			dev_err(&pdev->dev, "no device match found\n");
-			return -ENODEV;
-		}
-		set_dev_node(&pdev->dev, acpi_get_node(ACPI_HANDLE(&pdev->dev)));
+#ifdef CONFIG_ACPI
+	else
+		fabric = tegra234_cbb_acpi_get_fabric(&pdev->dev);
+#endif
+
+	if (!fabric) {
+		dev_err(&pdev->dev, "no device match found\n");
+		return -ENODEV;
 	}
 
 	cbb = devm_kzalloc(&pdev->dev, sizeof(*cbb), GFP_KERNEL);
@@ -1851,7 +1855,7 @@ static struct platform_driver tegra234_cbb_driver = {
 	.driver = {
 		.name = "tegra234-cbb",
 		.of_match_table = tegra234_cbb_dt_ids,
-		.acpi_match_table = tegra241_cbb_acpi_ids,
+		.acpi_match_table = ACPI_PTR(tegra241_cbb_acpi_ids),
 		.pm = &tegra234_cbb_pm,
 	},
 };
-- 
2.55.0

